File: scalaLearning/scalaActor/HelloActor.scala
import scala.actors.Actor

/**
 * Created by Variant on 16/3/30.
 */

object First_Actor extends Actor{
  def act(): Unit ={
    println(Thread.currentThread().getName)
    for (i <- 1 to 10){
      println("Step:" + i)
      Thread.sleep(1000)
    }
  }
}

object Second_Actot extends Actor{
  def act(): Unit = {
    println(Thread.currentThread().getName)
    for (i <- 1 to 10){
      println("Step :" + i)
      Thread.sleep(1000)
    }
  }
}

object HelloActor {
  def main(args: Array[String]) {
    First_Actor.start
    Second_Actot.start
  }
}

File: scalaLearning/Akka/src/main/java/akka/dt/app/java/HelloActor.java
package akka.dt.app.java;

import akka.actor.ActorRef;
import akka.actor.ActorSystem;
import akka.actor.Props;
import akka.dt.app.java.actors.MasterActor;
import akka.dt.app.java.messages.Result;

/**
 * Created by Variant on 16/4/9.
 */
public class HelloActor {

    public static void main(String[] args) throws Exception{
        ActorSystem _system = ActorSystem.create("HelloAkka");

        ActorRef master  =_system.actorOf(new Props(MasterActor.class),"master");

        master.tell("Hi! my name is Rockey. I am so happy to be here ");
        master.tell("Today,I'm going to read an article for you");
        master.tell("I sincerelly hope you can like that");

        Thread.sleep(500);

        master.tell(new Result());

        Thread.sleep(500);

        _system.shutdown();
    }
}
